source/dem/distributions.cc

```diff
@@ -68,3 +68,59 @@ UniformDistribution::find_max_diameter()
 {
   return this->diameter_values;
 }
+
+HistogramDistribution::HistogramDistribution(
+  const std::vector<double> &d_list,
+  const std::vector<double> &d_probabilities)
+{
+  diameter_hist_values = d_list;
+  std::vector<double> cumulative_probability_vector;
+  cumulative_probability_vector.reserve(d_probabilities.size());
+
+  double cumulative_value;
+  for (double i_probability : d_probabilities)
+    {
+      cumulative_value += i_probability;
+      cumulative_probability_vector.push_back(cumulative_value);
+    }
```

Comment:
I would use a typical `cummulative_probability_vector[i] = cummulative_value` instead of push_back since you know the size already.

Reply:
Ok, I understand what you are saying. Both are equivalent in this case (Pretty sure... we could ask)

Memory is allocated once with my implementation. The only difference (with reserve) is that memory is allocated but the size of the vector is still zero, so the vector can grow without re-allocating memory (what we don't want) 

If you do : 
```
std::vector<double> cumulative_probability_vector(d_probabilities.size() );  
// Here it's a vector of size d_probabilities.size() filled with zeros.
  double cumulative_value;
  for (unsigned int i = 0; i < d_probabilities.size() ; i++ )
    {
      cumulative_value += d_probabilities[i];
      cumulative_probability_vector[i] =  cumulative_value ;
    }
```
Memory is allocated once, but you change the values in the vector

Reply:
Ok I see! Thanks for the explaination :)

Reply:
I spoke with Bruno Bless and both are equivalent. :)